frontend/src/pages/RegisterInvestor.tsx

```diff
@@ -0,0 +1,245 @@
+      newErrors.confirmPassword = "Паролі не співпадають. Будь ласка, введіть однакові паролі в обидва поля";
+
+    if (!formData.lastName) newErrors.lastName = "Не ввели прізвище";
+    if (!formData.firstName) newErrors.firstName = "Не ввели ім’я";
+    if (!formData.representing.length) newErrors.representing = "Виберіть кого ви представляєте";
+    if (!formData.entityType.length) newErrors.entityType = "Виберіть тип суб’єкта";
+    if (!formData.minInvestment) newErrors.minInvestment = "Не ввели мінімальну інвестицію";
+    if (!formData.maxInvestment) newErrors.maxInvestment = "Не ввели максимальну інвестицію";
+
+    return newErrors;
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleCheckboxChange = (field: string, value: string) => {
+    setFormData((prev) => {
+      const updated = prev[field as keyof typeof prev] as string[];
+      if (updated.includes(value)) {
+        return { ...prev, [field]: updated.filter((v) => v !== value) };
+      } else {
+        return { ...prev, [field]: [...updated, value] };
+      }
+    });
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setErrors({});
+    setSuccessMessage("");
+
+    const validationErrors = validate();
+    if (Object.keys(validationErrors).length > 0) {
+      setErrors(validationErrors);
+      return;
+    }
+
+    try {
+      const response = await fetch("/api/auth/register/", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          email: formData.email,
+          password: formData.password,
+          role: formData.role,
+          company_name: formData.companyName,
+          investment_range_min: parseFloat(formData.minInvestment),
+          investment_range_max: parseFloat(formData.maxInvestment),
+        }),
+      });
+
+      const data = await response.json();
+
+      if (response.status === 400 && data.email) {
+        setErrors({ email: data.email[0] });
+      } else if (response.status === 201) {
+        setSuccessMessage(
+          "Реєстрація майже завершена. На зазначену вами електронну пошту відправлено листа. Будь ласка, перейдіть за посиланням з листа для підтвердження вказаної електронної адреси. Не отримали листа? Надіслати ще раз."
+        );
+        setFormData({
+          companyName: "",
+          email: "",
+          password: "",
+          confirmPassword: "",
+          lastName: "",
+          firstName: "",
+          representing: [],
+          entityType: [],
+          minInvestment: "",
+          maxInvestment: "",
+          role: "investor",
+        });
+      }
+    } catch {
+      setErrors({
+        general:
+          "Помилка активації. Під час активації сталась помилка. Спробуйте ще раз або зв'яжіться з підтримкою.",
+      });
+    }
```

Comment:
The catch block uses a bare catch without capturing the error, which swallows the thrown error and makes debugging difficult. Capture and log the error before setting the user-facing message.
```
} catch (e) {
  console.error("Register error:", e);
  setErrors({ general: "Помилка активації. ... зв'яжіться з підтримкою." });
}
```